src/core/control/shaperecognizer/ShapeRecognizer.cpp

```diff
@@ -242,13 +244,26 @@ void ShapeRecognizer::optimizePolygonal(const Point* pt, int nsides, int* breaks
     }
 }
 
+auto ShapeRecognizer::isStrokeLargeEnough(Stroke* stroke, double strokeMinSize) -> bool {
+    if (stroke->getPointCount() < 3) {
+        return false;
+    }
+
+    auto rect = stroke->getSnappedBounds();
+    if (rect.width < strokeMinSize && rect.height < strokeMinSize) {
+        return false;
+    }
+
+    return true;
```

Comment:
nit: can be simplified to
```suggestion
    return rect.width >= strokeMinSize || rect.height >= strokeMinSize;
```

Reply:
It looks like this only checks the axis-aligned bounding box. If someone sets a large minimum and draws diagonal lines, will there be a noticeable or awkward difference compared to drawing horizontal or vertical lines?

Reply:
Hm yeah there is no difference between drawing diagonal lines and completely vertical or horizontal lines because the condition only activates when both axis are smaller than the `strokeMinSize`. I don't think it'll be unnatural unless I'm not understanding your question correctly

Reply:
About the nit - I'm not sure if the simplification helps readability. It continues the pattern of the previous if-statement and the `<` makes it directly obvious that the width and lengths must both be within the bounding box - the suggested changes make it a bit less clear and there is a little more thinking involved. And it might be a bit of a stretch, but it might be more error prone when adding a conditional in the future, as the conditional would have to be DeMorganized again. But if you wish, shall I still make those changes?

Reply:
> Hm yeah there is no difference between drawing diagonal lines and completely vertical or horizontal lines because the condition only activates when both axis are smaller than the `strokeMinSize`. I don't think it'll be unnatural unless I'm not understanding your question correctly

It might be more natural to use:
```return std::hypot(rect.width, rect.height) >= strokeMinSize```
(length of the diagonal of the rectangle)